caddyconfig/caddyfile/lexer_test.go

```diff
@@ -285,6 +285,18 @@ EOF same-line-arg
 		},
 		{
 			input: []byte(`heredoc <<EOF
+EOF
+	HERE same-line-arg
+	`),
+			expected: []Token{
+				{Line: 1, Text: `heredoc`},
+				{Line: 1, Text: ``},
+				{Line: 3, Text: `HERE`},
+				{Line: 3, Text: `same-line-arg`},
+			},
+		},
+		{
+			input: []byte(`heredoc <<EOF
```

Comment:
Can you also add the same test case found by the fuzzer?

Reply:
Sorry, I didn't understand what you said above.
What is the same test case found by the fuzzer?

Reply:
Maybe I misunderstood. Is this PR resolving #6058?

Reply:
That's another issue.
#6061 will fix the performance regression in #6058.

Reply:
Noted.